esda/getisord.py

```diff
@@ -518,5 +557,88 @@ def by_col(
+    elif (not zero_diagonal) & (star is True):
+        weights = weights
+    # Want zero diagonal and have it
+    elif zero_diagonal & (star is False):
+        weights = weights
+    # Want nonzero diagonal and do not have it
+    elif zero_diagonal & (star is True):
+        # if the input is binary or requested transform is binary,
+        # set the diagonal to 1.
+        if transform.lower() == "b" or weights.transform.lower() == "b":
+            weights = fill_diagonal(weights, 1)
+        # if we know the target is row-standardized, use the row max
+        # this works successfully for effectively binar but "O"-transformed input
+        elif transform.lower() == "r":
+            # This warning is probably better as a documentation instead of a warning.
+            # warnings.warn(
+            #     "Gi* requested, but no weights are on the diagonal and"
+            #     " no default value supplied to star. Assuming that the"
+            #     " self-weight is equivalent to the maximum weight in the"
+            #     " row. To use a different default (like, .5), set `star=.5`,"
+            #     " or use libpysal.weights.fill_diagonal() to set the diagonal"
+            #     " values of your weights matrix and use `star=None` in Gi_Local."
+            # )
+            weights = fill_diagonal(
+                weights, np.asarray(adj_matrix.max(axis=1).todense()).flatten()
+            )
+    else:  # star was something else, so try to fill the weights with it
+        try:
+            weights = fill_diagonal(weights, star)
+        except:
+            raise TypeError(
+                f"Type of star ({type(star)}) not understood."
+                f" Must be an integer, boolean, float, or numpy.ndarray."
+            )
+    star = (weights.sparse.diagonal() > 0).any()
+    weights.transform = transform
+
+    return weights, star
+
+if __name__ is '__main__':
+    import geopandas, numpy, esda, importlib
+    import matplotlib.pyplot as plt
+    from libpysal import weights, examples
+
+    df = geopandas.read_file(examples.get_path('NAT.shp'))
+
+    w = weights.Rook.from_dataframe(df)
+
+    for transform in ('r', 'b'):
+      for star in (True, False):
+        test = esda.getisord.G_Local(df.GI89, w, transform=transform, star=star)
+        out = test._calc2()
+        (statistic, expected_value, 
+         expected_variance, z_scores, 
+         empirical_mean, empirical_variance) = out
+
+        numpy.testing.assert_allclose(statistic, test.Gs)
+        numpy.testing.assert_allclose(expected_value, test.EGs)
+        numpy.testing.assert_allclose(expected_variance, test.VGs)
+        numpy.testing.assert_allclose(z_scores, test.Zs)
+        numpy.testing.assert_allclose(empirical_mean, test.yl_mean)
+        numpy.testing.assert_allclose(empirical_variance, test.s2)
+
+    # Also check that the None configuration works
+    test = esda.getisord.G_Local(df.GI89, w, star=None)
```

Comment:
This will fail right now, but the intent is to eventually test the `star==None`, which should be equivalent to `star==False` for this configuration. 
Then, fill the weights diagonal with 1 and check that None returns results akin to star==True.